src/main/java/Parser.java

```diff
@@ -0,0 +1,122 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+// DEALS WITH MAKING SENSE OF USER'S COMMAND
+public class Parser {
+
+    protected Storage storage;
+    protected Ui ui;
+    protected TaskList tasks;
+
+    public Parser(Storage storage, Ui ui, TaskList tasks) {
+        this.storage = storage;
+        this.ui = ui;
+        this.tasks = tasks;
+    }
+
+    static String parseCommandType(String command) {
+        String[] tokenizedCommand = command.split(" ");
+        return tokenizedCommand[0];
+    }
+
+    static int parseTaskIndex(String command) {
+        String[] tokenizedCommand = command.split(" ");
+        return Integer.parseInt(tokenizedCommand[1]);
+    }
+
+    static String parseTodoDescription(String command) {
+        return command.substring(5);
+    }
+
+    static String parseDeadlineDescription(String command) {
+        return command.substring(9).split(" /by ")[0];
+    }
+
+    static LocalDate parseDeadlineDate(String command) {
+        return LocalDate.parse(command.substring(9).split(" /by ")[1]);
```

Comment:
Maybe you could split them into 3 different lines to make the code easier to understand